pkg/spoke/controllers/manifestwork_controller.go

```diff
@@ -425,3 +405,57 @@ func buildAppliedStatusCondition(err error) workapiv1.StatusCondition {
 		Message: "Apply manifest complete",
 	}
 }
+
+func buildManifestResourceMeta(index int, object runtime.Object, restMapper *resource.Mapper) (resourceMeta workapiv1.ManifestResourceMeta, err error) {
+	resourceMeta = workapiv1.ManifestResourceMeta{
+		Ordinal: int32(index),
+	}
+
+	if object == nil {
+		return resourceMeta, err
+	}
+
+	var gvk schema.GroupVersionKind
+	var namespace, name string
+
+	switch objectType := object.(type) {
+	case *unstructured.Unstructured:
```

Comment:
this case doesn't collapse into the resourcehelper.GuessObjectGroupVersionKind(objectType) below?

Reply:
I tried and found GuessObjectGroupVersionKind() works for Unstructured object. I'll refine the code in a followup pr.